libvast/vast/detail/concepts.hpp

```diff
@@ -41,4 +56,31 @@ concept unsigned_integral = integral<T> && std::is_unsigned_v<T>;
 template <class T>
 concept signed_integral = integral<T> && std::is_signed_v<T>;
 
+struct any_callable {
+  template <class... Ts>
+  void operator()(Ts&&...);
+};
+
+/// Inspectables
+template <class T>
+concept is_inspectable = requires(detail::any_callable& i, T& x) {
+  {inspect(i, x)};
+};
+
+/// push_back
+template <class C>
+concept has_insert = requires(C xs, typename C::value_type x) {
+  xs.insert(x);
+};
+
+template <class C>
+concept has_push_back = requires(C xs, typename C::value_type x) {
+  xs.push_back(x);
+};
+
+template <class T>
+concept monoid = requires(T x, T y) {
```

Comment:
Can we add some unit tests for these (non std) concepts, please?

Reply:
I have to admit I'm not sure how to unit test meta programming utilities.

Reply:
`static_assert`s are how you can test them. For example, for monoid, here are some tests that I'd recommend: https://godbolt.org/z/Ps659sKb5

Reply:
I like those. I don't remember where, but didn't you recently introduce some tests like this as well, @JoeLoser?

Reply:
Yeah. When I started introducing concepts, I added unit tests along the way. These started out in https://github.com/tenzir/vast/pull/1791.

Reply:
Thanks for the point. @tobim, maybe add a few concept tests there?